Added offset key to kubernetes logs the same as in file.rs (#22203)
vectordotdev/vector: changelog.d/add_offset_key_kubernetes_logs.enhancement.md

> The `kubernetes_logs` source now adds an offset key the same as file.rs.

Reviewer: ```suggestion
The `kubernetes_logs` source now support an `offset_key` like the `file` source does.
```

Author: @pront thanks for noticing. I removed it and commit the changes
